arch/arm/soc/st_stm32/common/Kconfig.defconfig.series

```diff
@@ -72,4 +72,11 @@ config USB_DC_STM32
 
 endif # USB
 
+if RTC
+
+config RTC_STM32
```

Comment:
The new patch doesn't support STM32F1 series, right? Then, it should not be default y for all STM32 series.

Reply:
You are right. The driver doesn't supoort STM32F1 series. I will update the PR.

Reply:
Well, I think it makes sense to put it in common.

RTC should not be enabled if not implemented for one series (F1 in the case you mention).
RTC_STM32 should be used if RTC is enabled, this is true for all STM32 series (would not be the case if 2 versions of STM32 RTC driver were available as a choice to the user).

This is not different from any other driver in arch/arm/soc/st_stm32/common/Kconfig.defconfig.series

Reply:
```
if RTC && !SOC_SERIES_STM32F1X
        def_bool y
endif # RTC && !SOC_SERIES_STM32F1X
```
@erwango Does the above works for you? It makes clear that the RTC_STM32 driver doesn't support STM32F1 right now.

Reply:
But then what for other series? Is there something different for RTC for STM32F1 different vs RTC driver for other series not supported yet (F0/F2/F7/L0)?
Besides, use of negative is complex to maintain and should be updated each time a new series is added.

@jli157 : To me, current use and role of `common/Kconfig.defconfig.series` is to enable STM32 versions of drivers not to do any gating.
If one enables RTC on F1, he'll face issues because it is not available yet. Such as the one enabling SPI for a recently introduced series.

Reply:
@erwango I get you point, I have updated the PR.

Reply:
Thanks. Completing the review right now